src/Core/Injector/Injector.php

```diff
@@ -851,21 +851,26 @@ public function registerService($service, $replace = null)
      * by the inject
      *
      * @param string $name The name to unregister
+     * @param bool $flushSpecs Set to true to clear spec for this service
      * @return $this
      */
-    public function unregisterNamedObject($name)
+    public function unregisterNamedObject($name, $flushSpecs = false)
```

Comment:
I've got mixed feelings about this. Could we just flush specs in all cases? This method seems to be only used for edge-cases such as tests.